internal/services/oracle/oracle_autonomous_database_clone_data_source.go

```diff
@@ -0,0 +1,366 @@
+		},
+
+		"mtls_connection_required": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"national_character_set": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"subnet_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"virtual_network_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"lifecycle_state": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"private_endpoint": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"private_endpoint_ip": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"service_console_url": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"sql_web_developer_url": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"time_created": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"oci_url": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (AutonomousDatabaseCloneDataSource) ModelObject() interface{} {
+	return &AutonomousDatabaseCloneDataSourceModel{}
+}
+
+func (AutonomousDatabaseCloneDataSource) ResourceType() string {
+	return "azurerm_oracle_autonomous_database_clone"
+}
+
+func (AutonomousDatabaseCloneDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			client := metadata.Client.Oracle.OracleClient.AutonomousDatabases
+
+			var model AutonomousDatabaseCloneDataSourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := autonomousdatabases.NewAutonomousDatabaseID(subscriptionId,
+				model.ResourceGroupName,
+				model.Name)
+
+			resp, err := client.Get(ctx, id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			model.Name = id.AutonomousDatabaseName
+			model.ResourceGroupName = id.ResourceGroupName
+
+			if m := resp.Model; m != nil {
+				model.Location = m.Location
+				model.Tags = pointer.From(m.Tags)
+
+				// Verify this is actually a clone database
+				props, ok := m.Properties.(autonomousdatabases.AutonomousDatabaseCloneProperties)
+				if !ok {
+					return fmt.Errorf("%s is not a clone type autonomous database", id)
+				}
+
+				// Clone-specific properties
+				model.CloneType = string(props.CloneType)
+				model.SourceId = props.SourceId
+				model.IsReconnectCloneEnabled = pointer.From(props.IsReconnectCloneEnabled)
+				model.IsRefreshableClone = pointer.From(props.IsRefreshableClone)
+				model.TimeUntilReconnectClone = pointer.From(props.TimeUntilReconnectCloneEnabled)
+
+				if props.RefreshableModel != nil {
+					model.RefreshableModel = string(*props.RefreshableModel)
+				}
+				if props.RefreshableStatus != nil {
+					model.RefreshableStatus = string(*props.RefreshableStatus)
+				}
```

Comment:
We have a helper for this

```suggestion
				model.RefreshableModel = pointer.FromEnum(props.RefreshableModel)
				model.RefreshableStatus = pointer.FromEnum(props.RefreshableStatus)
```